src/components/QueryTagManager/QueryTagManager.tsx

```diff
@@ -75,29 +75,31 @@ const QueryTagManager: FC<Props> = ({ tags, onAddTag, onRemoveTag }) => {
 
   const keyboardHandler = useCallback(
     (e: KeyboardEvent) => {
+      if (!tagsRef.current?.length) return;
       switch (e.keyCode) {
         case KEYBOARD_KEYS.ENTER:
-          const { value } = tagsHints[indexRef.current];
-          if (!tags.includes(value)) {
+          const { value } = tagsRef.current[indexRef.current];
+          if (value && !tags.includes(value)) {
             onAddTag(value);
           }
 
           inputRef.current.value = '';
           setDropdownVisibility(false);
+          setTagsHint(null);
           break;
         case KEYBOARD_KEYS.UP:
           if (indexRef.current > 0) {
             setIndex(indexRef.current - 1);
           }
           break;
         case KEYBOARD_KEYS.DOWN:
-          if (indexRef.current < tagsHints.length - 1) {
+          if (indexRef.current < tagsRef.current.length - 1) {
             setIndex(indexRef.current + 1);
           }
           break;
       }
     },
-    [tagsHints, tags]
+    [tagsRef, tags]
```

Comment:
`tagsRef` reference was caught in the **closure** so I believe there is no need to add it to dependences array.